services/QuillLMS/app/models/unit_activity.rb

```diff
@@ -122,7 +124,8 @@ def self.get_classroom_user_profile(classroom_id, user_id)
           ua.activity_id,
           MAX(acts.updated_at) AS act_sesh_updated_at,
           ua.order_number,
-          ua.due_date,
+          CASE WHEN teachers.time_zone IS NOT NULL THEN ua.due_date at time zone teachers.time_zone ELSE ua.due_date END AS due_date,
+          CASE WHEN teachers.time_zone IS NOT NULL THEN ua.publish_date at time zone teachers.time_zone ELSE ua.publish_date END AS publish_date,
```

Comment:
It feels like this particular pattern could get cumbersome, but I think that there's probably not a simpler approach in cases where we're writing raw SQL rather than going through ActiveRecord.  So let's go with this for now, but keep an eye out for how often we have to do this sort of thing.